src/test_application/OSPMagnum.cpp

```diff
@@ -160,76 +162,165 @@ osp::active::ActiveScene& OSPMagnum::scene_create(std::string && name)
 void testapp::config_controls(OSPMagnum& rOspApp)
 {
     // Configure Controls
+    //Load toml
+    auto data = toml::parse("settings.toml");
+    try 
+    {
+        osp::UserInputHandler& rUserInput = rOspApp.get_input_handler();
+        for (const auto& [k, v] : data.as_table())
+        {
+            std::cout << "Parsing " << k << std::endl;
+            std::string primary = toml::find(v, "primary").as_string();
+            std::vector<osp::ButtonVarConfig> controls = parse_control(primary);
+
+            std::string secondary = toml::find(v, "secondary").as_string();
+            std::vector<osp::ButtonVarConfig> secondaryKeys = parse_control(secondary);
+
+            controls.insert(controls.end(), secondaryKeys.begin(), secondaryKeys.end());
+
+            bool holdable = toml::find(v, "holdable").as_boolean();
+
+            //Add user input
+            rUserInput.config_register_control(k, holdable, controls);
```

Comment:
This is the last place where you're referencing "controls". 
As such you'll need to make sure you use std::move() on it to tell the compiler you're done with it.
Sadly, it can't detect that itself.
See here for a bit more info: https://stackoverflow.com/questions/45763943/compiler-deduction-of-rvalue-references-for-variables-going-out-of-scope